web-ui/src/components/reports-section/CheckinReport.jsx

```diff
@@ -1,36 +1,188 @@
+          const labelProps = {};
+          if (isStepOptional(index)) {
+            labelProps.optional = (
+              <Typography variant="caption">Optional</Typography>
+            );
+          }
+          if (isStepSkipped(index)) {
+            stepProps.completed = false;
+          }
+          return (
+            <Step key={label} {...stepProps}>
+              <StepLabel {...labelProps}>{label}</StepLabel>
+            </Step>
+          );
+        })}
+      </Stepper>
+    </Box>
+  );
+}
+
 import './CheckinReport.css';
 
+const propTypes = {
+  closed: PropTypes.bool,
+  pdl: PropTypes.shape({
+    name: PropTypes.string,
+    id: PropTypes.string,
+    members: PropTypes.array,
+    workEmail: PropTypes.string,
+    title: PropTypes.string
+  }),
+  planned: PropTypes.bool
+};
+
 const CheckinsReport = ({ closed, pdl, planned }) => {
   const { state } = useContext(AppContext);
-  const { name, id, members, workEmail } = pdl;
+  const [expanded, setExpanded] = useState(true);
+  const [statusForPeriod, setStatusForPeriod] = useState(
+    /** @type CheckinStatus */ ('Not Started')
+  );
+
+  const { name, id, members, workEmail, title } = pdl;
+
+  const handleExpandClick = () => setExpanded(!expanded);
 
   const getCheckinDate = checkin => {
     if (!checkin || !checkin.checkInDate) return;
     const [year, month, day, hour, minute] = checkin.checkInDate;
     return new Date(year, month - 1, day, hour, minute, 0);
   };
 
+  /**
+   * Determine the status of the check-ins for a member.
+   * @param {Object} params - The parameters object.
+   * @param {Checkin[]} params.checkins - Checkins for a member.
+   * @returns {CheckinStatus} The status of check-ins.
+   */
+  const statusForPeriodByMember = ({ checkins = [] }) => {
+    const now = new Date();
+    if (checkins.length === 0) return 'Not Started';
+    const completed = checkins.filter(checkin => checkin.completed);
+    if (completed.length === checkins.length) return 'Done';
+    const inProgress = checkins.filter(
+      checkin => !checkin.completed && getCheckinDate(checkin) < now
+    );
+    if (inProgress.length > 0) return 'Open';
+    return 'Not Started';
+  };
+
+  /**
+   * Get the date of the last check-in.
+   * @param {Checkin[]} checkins - Check-ins for a member.
+   * @returns {Date} The date of the last check-in.
+   */
+  const getLastCheckinDate = checkins => {
+    if (checkins.length === 0) return;
+    return checkins.reduce((acc, checkin) => {
+      const checkinDate = getCheckinDate(checkin);
+      return checkinDate > acc ? checkinDate : acc;
+    }, new Date(0));
+  };
+
+  /**
+   * Determine the status of the check-ins for a PDL.
+   * @param {Object} params - The parameters object.
+   * @param {Array} params.members - Members of the PDL.
+   * @returns {CheckinStatus} The status of check-ins.
+   */
+  const statusForPeriodByPDL = ({ members = [] }) => {
+    // const debug = ['Done', 'In Progress', 'Not Started'][0];
```

Comment:
Finished with this line?

Reply:
Yes. You got me into the review before I was ready. This is cleaned up now.